projects/storefrontlib/src/shared/components/form/form-errors/form-errors.component.ts

```diff
@@ -28,16 +35,17 @@ export class FormErrorsComponent {
   @Input()
   set control(control: FormControl) {
     this._control = control;
+    this.setErrors();
+  }
 
-    this.errors$ = control?.statusChanges.pipe(
-      startWith({}),
-      map(() => control.errors || {}),
-      map((errors) =>
-        Object.entries(errors)
-          .filter((error) => error[1])
-          .map((error) => error[0])
-      )
-    );
+  /**
+   * The name of the control is enough
+   * @param controlName
+   */
+  @Input()
+  set controlName(controlName: string) {
```

Comment:
I've checked once more why we can't use this component with a static controlName reference, as I believe this is quite a drop in dx. 

```html
<form-errors controlName="email"></form-errors>
```

I believe this is caused by the fact that at the time the static input is set, the form container/control  is not yet initialised. You'd therefor need to move the logic to a lifecycle hook. 

To support dynamic forms, I'd recommend to use the `OnChanges` hook, and validate whether the controlName or control are changed. With that, you can further simplify the controller: 
- use simple inputs (no getters/setters needed)
- use the public `control` only (drop private _control), and set the control only if the controlName is passed in